projects/opal-frontend-common/directives/capitalisation/capitalisation.directive.ts

```diff
@@ -19,9 +19,12 @@ export class CapitalisationDirective {
     const target = event.target as HTMLInputElement;
     const value = target.value;
 
-    if (value) {
-      const capitalisedValue = this.utilsService.upperCaseAllLetters(value);
+    const capitalisedValue = this.utilsService.upperCaseAllLetters(value);
+    this.renderer.setProperty(target, 'value', capitalisedValue);
+
+    if (value !== capitalisedValue) {
       this.renderer.setProperty(target, 'value', capitalisedValue);
+      target.dispatchEvent(new Event('input', { bubbles: true }));
```

Comment:
- Only sets the value and fires an input event if actual change is detected

```suggestion
    const capitalisedValue = this.utilsService.upperCaseAllLetters(value);

    if (value !== capitalisedValue) {
      this.renderer.setProperty(target, 'value', capitalisedValue);
      target.dispatchEvent(new Event('input', { bubbles: true }));
    }
```